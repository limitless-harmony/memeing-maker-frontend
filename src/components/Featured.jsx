import React, { Component } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { calculateRem } from 'styles';
import { getFeatured } from 'actions/meme';
import MemeCard from 'components/MemeCard';
import Button from 'components/Button';
import { buttonBorder, black } from 'styles/colors';

export class Featured extends Component {
  async componentDidMount() {
    const { actions } = this.props;
    await actions.getFeatured();
  }

  render() {
    const { memes } = this.props;
    return (
      <StyledFeatured>
        {memes.length ? (
          <>
            <SectionHeading>This Weekâ€™s Featured Memes</SectionHeading>
            <Container>
              {memes.map(meme => (
                // eslint-disable-next-line no-underscore-dangle
                <Meme key={meme._id}>
                  <MemeCard
                    small
                    src={meme.image}
                    topText={meme.topText}
                    bottomText={meme.bottomText}
                  />
                </Meme>
              ))}
            </Container>
            <Button color={buttonBorder} outline rounded>
              <ButtonText to="/login">More</ButtonText>
            </Button>
          </>
        ) : (
          <SectionHeading>No featured meme for this week!</SectionHeading>
        )}
      </StyledFeatured>
    );
  }
}

const StyledFeatured = styled.div`
  margin: ${calculateRem(10)} auto ${calculateRem(40)};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
`;

const Container = styled.div`
  margin: ${calculateRem(10)} auto ${calculateRem(40)};
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;
  text-align: center;
`;

const Meme = styled.div`
  width: 48.5%;
`;

const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;

const ButtonText = styled(Link)`
  color: ${black};
  text-decoration: none;
`;

const mapStateToProps = state => ({
  memes: state.meme.featured,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ getFeatured }, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Featured);
