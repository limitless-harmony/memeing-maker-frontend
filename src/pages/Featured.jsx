import React, { Component } from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import Masonry from 'components/Masonry';
import MemeCard from 'components/MemeCard';
import Button from 'components/Button';
import Yes from 'assets/images/yes.png';
import Sense from 'assets/images/sense.jpg';
import Duh from 'assets/images/duh.jpg';
import { buttonBorder, black } from 'styles/colors';

class Featured extends Component {
  greet = () => {};

  render() {
    return (
      <StyledFeatured>
        <Slogan>Make meaning. Share meaning.</Slogan>
        <Title>Memeing Maker</Title>
        <SectionHeading>This Weekâ€™s Featured Memes</SectionHeading>
        <Masonry>
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
          <MemeCard
            imageUrl={Yes}
            topText="A top text"
            bottomText="A bottom text"
          />
          <MemeCard
            imageUrl={Sense}
            topText="When you know"
            bottomText="And you know that you know"
          />
          <MemeCard
            imageUrl={Duh}
            topText="You think you know me"
            bottomText="Duh! You have no idea. I am like the wind!"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/320/240"
            topText="You think you know me"
            bottomText="Duh! You have no idea. I am like the wind!"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
          <MemeCard
            imageUrl="https://loremflickr.com/240/320"
            topText="Randomly generated"
            bottomText="New every time"
          />
        </Masonry>
        <Button color={buttonBorder} outline rounded>
          <ButtonText>More</ButtonText>
        </Button>
      </StyledFeatured>
    );
  }
}

const StyledFeatured = styled.div`
  margin: ${calculateRem(40)} auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
`;

export const Title = styled.div`
  margin: 0 auto ${calculateRem(60)};
  line-height: ${calculateRem(43)};
  font-size: ${calculateRem(36)};
`;

export const Slogan = styled.div`
  margin: 0 auto;
`;
export const SectionHeading = styled.div`
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(16)} auto;
`;

const ButtonText = styled.div`
  color: ${black};
`;

export default Featured;
