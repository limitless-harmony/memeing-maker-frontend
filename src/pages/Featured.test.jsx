import React from 'react';
import { shallow } from 'enzyme';
import Featured from 'pages/Featured';

describe('Testing the Featured page component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Featured />);
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
