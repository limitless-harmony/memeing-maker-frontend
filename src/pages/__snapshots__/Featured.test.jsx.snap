// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Featured page component renders without crashing 1`] = `
<styled.div>
  <styled.div>
    Make meaning. Share meaning.
  </styled.div>
  <styled.div>
    Memeing Maker
  </styled.div>
  <styled.div>
    This Weekâ€™s Featured Memes
  </styled.div>
  <Masonry>
    <MemeCard
      src="meme.png"
    />
    <MemeCard
      src="meme1.png"
    />
    <MemeCard
      src="meme2.png"
    />
    <MemeCard
      src="meme3.png"
    />
    <MemeCard
      src="meme4.png"
    />
  </Masonry>
  <Button
    color="#979797"
    outline={true}
    rounded={true}
  >
    <styled.div>
      More
    </styled.div>
  </Button>
</styled.div>
`;
